web/pgadmin/preferences/static/js/components/PreferencesComponent.jsx

```diff
@@ -546,62 +578,74 @@ export default function PreferencesComponent({ ...props }) {
     return requires_refresh;
   }
 
-  function save(save_data, data) {
+  function save(save_data, data, layout_changed=false) {
     api({
       url: url_for('preferences.index'),
       method: 'PUT',
       data: save_data,
     }).then(() => {
-      let requiresTreeRefresh = save_data.some((s)=>{
-        return (
-          s.name=='show_system_objects' || s.name=='show_empty_coll_nodes' ||
-          s.name.startsWith('show_node_') || s.name=='hide_shared_server' ||
-          s.name=='show_user_defined_templates'
-        );
-      });
-      let requires_refresh = false;
-      for (const [key] of Object.entries(data.current)) {
-        let pref = preferencesStore.getPreferenceForId(Number(key));
-        requires_refresh = checkRefreshRequired(pref, requires_refresh);
-      }
-
-      if (requiresTreeRefresh) {
-        pgAdmin.Browser.notifier.confirm(
-          gettext('Object explorer refresh required'),
-          gettext(
-            'An object explorer refresh is required. Do you wish to refresh it now?'
-          ),
-          function () {
-            pgAdmin.Browser.tree.destroy().then(
-              () => {
-                pgAdmin.Browser.Events.trigger(
-                  'pgadmin-browser:tree:destroyed', undefined, undefined
-                );
-                return true;
-              }
+      // If layout is changed then only refresh the object explorer.
+      if (layout_changed) {
+        pgAdmin.Browser.tree.destroy().then(
```

Comment:
Why do we need to refresh the tree on layout change?

Reply:
This is must as we are disconnecting from all servers on Layout changes